app/src/main/scala/org/alephium/app/ServerUtils.scala

```diff
@@ -1917,6 +1970,53 @@ object ServerUtils {
     }
   }
 
+  def getPositiveAlphRemainderOrFail(
+      fromUnlockScript: UnlockScript,
+      inputs: AVector[AssetOutputInfo],
+      outputs: AVector[TxOutputInfo],
+      gasPrice: GasPrice
+  ): Either[String, U256] = {
+    if (inputs.isEmpty || outputs.isEmpty) {
+      Left("Both inputs and outputs must be specified")
+    } else {
+      val inputAmounts = inputs.map(_.output.amount)
+      val totalInputValue =
+        inputAmounts.fold(Option(U256.Zero)) { case (acc, n) =>
+          acc.flatMap(_.add(n))
+        }
+      val outputAmounts = outputs.map(_.attoAlphAmount)
+      val totalOutputValue =
+        outputAmounts.fold(Option(U256.Zero)) { case (acc, n) =>
+          acc.flatMap(_.add(n))
+        }
+
+      (totalInputValue, totalOutputValue) match {
+        case (Some(iv), Some(ov)) if iv > ov =>
+          for {
+            gasBox <- GasEstimation.estimateWithInputScript(
+              fromUnlockScript,
+              inputs.length,
+              outputs.length + 1,                    // + 1 is for change utxo
+              AssetScriptGasEstimator.NotImplemented // Not P2SH
+            )
+            remainder <- UnsignedTransaction.calculateAlphRemainder(
+              inputAmounts,
+              outputAmounts,
+              gasPrice * gasBox
+            )
+          } yield remainder
+        case (None, Some(_)) =>
+          Left(s"Invalid total input value")
+        case (Some(_), None) =>
+          Left(s"Invalid total output value")
+        case (None, None) =>
+          Left(s"Invalid total input and output value")
+        case (Some(iv), Some(ov)) =>
+          Left(s"Total input value $iv is not enough for output value $ov and a gas fee")
+      }
+    }
+  }
```

Comment:
This method exists as part of "Fail Fast" approach so that we don't fail down at building 7th transaction but rather immediately as it would just confuse the api-user.